### DISCUSSION TITLE
Dynamic bandwidth - Help

### DISCUSSION THREAD

**Discussion by mas-oliveira** [esnet/iperf#1730] [General]
Hello guys,
I'm using iperf 3.7 (cJSON 1.5.2).
I wanted to know if it is possible to adjust the iperf tool to, instead of launching and during a duration the test uses the initial bandwidth, for instance, 1M, if there is any possibility to adjust, for instance, every 20 seconds the value, from 1M to 2M... and following this logic during 1000 seconds of the test.
Thanks a lot for the help.

**Comment by davidBar-On** [comment]
Hi, there is no such option.  You can use shell script, etc. to run the iperf3 client in a loop, using `-t 20` so each run will be 20 seconds, and set a different bitrate (using `-b`) for each run in the loop.

**Reply by mas-oliveira** [reply]
Thanks, David, that's the solution that I have at the moment, so I'll keep it. Since I'm running the iPerf with a Python script, the idea was, if possible, not to do this restart in order to keep the script working. But I think that I can move this script.
